### DISCUSSION TITLE
Labeling all required fields vs. just the optional fields

### DISCUSSION THREAD

**Discussion by pamelahu** [twilio-labs/paste#2050] [General] [answered]
Via Paste guidelines: 

> In a form where there are fewer optional fields than required, use an optional indicator to show users which fields they don’t have to fill out.

 This makes sense. I've also seen the argument for labeling all the required fields: [https://www.nngroup.com/articles/required-fields/](https://www.nngroup.com/articles/required-fields/)

What is the team's opinion on that?

**Comment by serifluous** [comment, marked as answer]
Hey @pamelahu , great question! We discussed Nielsen Norman's guidance in the original iteration of the form label design. After feedback from product designers, there were competing use cases where we'd want to indicate either all of the optional fields or all of the required fields. Most fields are required in Twilio products, and we didn't want to overload customers with too much noise. On the other hand, regulatory forms in Twilio have mostly optional fields, but we didn't want to dissuade customers from providing more information since it would make it easier for them to make their phone numbers compliant. 

We wrote the guidelines in that way to also allow teams and product designers to decide whether an optional or required indicator would best fit their use case.

If you're even more curious, you can take a look at more detailed rationale and research in the original brief! https://docs.google.com/document/d/19XgUJVBs7bfW1h-SjNfuAUNnx93hBABT1zQJftrCQ-M/edit#bookmark=id.dim5sx9jyxg6

**Reply by rosemaryluo** [reply]
Hello! Following up on this, do we have a defined visual indicator for optional?

**Reply by SiTaggart** [reply]
Not officially, no. We used to have a style, it was part of the original designs, but it never made it to engineering. Because of that, coupled with low use cases and Sarah's explanation above, I think we removed it from the official Paste component in Figma.
